firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexManager.java

```diff
@@ -83,12 +98,8 @@ public interface IndexManager {
   /** Returns the minimum offset for the given collection group. */
   IndexOffset getMinOffset(String collectionGroup);
 
-  /**
-   * Returns an index that can be used to serve the provided target. Returns {@code null} if no
-   * index is configured.
-   */
-  @Nullable
-  FieldIndex getFieldIndex(Target target);
+  /** Returns the type of index (if any) that can be used to serve the given target */
+  IndexStatus canServeUsingIndex(Target target);
```

Comment:
I think this is not a good name - it would be if we also passed the Index, but we don't. Maybe something like "hasIndex()"/"hasBackingIndex()"? 

Optional.